pkg/builders/s2i/builder.go

```diff
@@ -151,6 +151,29 @@ func (b *Builder) Build(ctx context.Context, f fn.Function, platforms []fn.Platf
 	}
 	defer os.RemoveAll(tmp)
 
+	funcignorePath := filepath.Join(f.Root, ".funcignore")
+	if _, err := os.Stat(funcignorePath); err == nil {
+		funcignore, err := os.Open(filepath.Join(f.Root, ".funcignore"))
+		if err != nil {
+			return fmt.Errorf("error opening funcignore file: %w", err)
+		}
+		defer funcignore.Close()
+
+		s2iignore, err := os.Create(filepath.Join(f.Root, ".s2iignore"))
```

Comment:
If the process panics, or otherwise fails to execute the `defer` statement's removal of `.s2iignore`, it may be left on disk after the process exits.

We may want this to create a symbolic link to `.funcignore` if it doesn't exist (and if S2I will still respect it), or use `Open` when creating the file with the "truncate" option so that it does't get an error that the file already exists.

Reply:
@lkingland I have updated it with symbolic link, PTAL